enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+  On OCP, kube-apiserver delegates the Route API to openshift-apiserver, which employs the library-go validation to validate Route objects (see <https://github.com/openshift/openshift-apiserver/blob/aac3dd5bf0547e928103a0f718ca104b1bb13930/pkg/route/apis/route/validation/validation.go>).
+  MicroShift does not use openshift-apiserver; instead, on MicroShift, kube-apiserver handles the Route API using the Route CRD. 
+    
+  However, kube-apiserver has an admission plugin for the Route CRD that employs the same library-go validation that openshift-apiserver uses (see <https://github.com/openshift/kubernetes/blob/15f19ea2dd700767e5337502aec753d2a6e26905/openshift-kube-apiserver/admission/customresourcevalidation/route/validate_route.go>).
+  Defining validation for the same API in both library-go and the OpenAPI schema introduces the risk that someone could inadvertently introduce discrepancies between the library-go validation and the OpenAPI schema, leading to buggy and potentially confusing behavior.  Care must be taken to avoid introducing such discrepancies.
+  Historical note: At one time, MicroShift relied on the OpenAPI schema for server-side validation of the Route CRD, and for this reason, kubebuilder validation tags were added to the
+  Route CRD (see <https://github.com/openshift/api/pull/1228>).
+
+  Since then, an admission plugin has been added to kube-apiserver that validates the Route CRD using the same
+  library-go validation on MicroShift that openshift-apiserver uses on OpenShift (see <https://github.com/openshift/kubernetes/pull/1416>).
+  As a result, the kubebuilder validation tags on the Route CRD are no longer required specifically for MicroShift; however, the kubebuilder validation tags are still required for client-side validation.
+
+#### Why don't you set/delete the headers on frontend public_ssl in the haproxy.config?
+  Answer: The connection is still encrypted in public_ssl, so it is not possible to examine or modify the HTTP session there. The fe_sni and fe_no_sni frontends terminate TLS.
+
+#### Can we add conditions for haproxy to check to set the value? For example: Can I set the value `"%[res.hdr(server)] if { req.hdr(user-agent) -m sub evil}"` for `x-frame-options` ?
+  Answer: You will be able to set the value however the if keyword won't be considered as a condition but a string by haproxy. 
+  For example: So, if you set x-frame-options: "%[res.hdr(server)] if { req.hdr(user-agent) -m sub evil}" as HTTP response then the response value served by 
+  will be `x-frame-options: Apache/2.4.37 (centos) if { req.hdr(user-agent) -m sub evil}` where we can see if condition is not evaluated as it is passed as a string to
+  haproxy.
+  
+  All values are escaped and quoted using the single quotes. Please check section `Alternative quoting strategies for haproxy.config` for more
+  details. So the value we get after escape and quote we get is a string. 
```

Comment:
This is not very clear.  I think the answer is "No."
```suggestion
  Answer: No.  You cannot set a conditional value.
  For example: If you set x-frame-options: "%[res.hdr(server)] if { req.hdr(user-agent) -m sub evil}" on an HTTP x-frame-options response header, then the response value 
  will be the literal:  `x-frame-options: Apache/2.4.37 (centos) if { req.hdr(user-agent) -m sub evil}`.  In other words, the condition is not evaluated.
  
```

Reply:
done